### PR TITLE
Trigger signals when child is added or removed. Add unit test.

### PR DESCRIPTION
**Pull request by lia-m99**

### CODE REVIEW

File: src/view/scene_model.hpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <project/prototypes.hpp>
+#include <unordered_map>
```
Comment (arsdever) [inline src/view/scene_model.hpp]: Please remove this line, until we figure out why the compiler doesn't find the header.

File: test/unit/object.cpp [hunk 2]
```diff
@@ -53,3 +53,30 @@ TEST(object, remove)
...
+        g::project::object::create("Child", nullptr);
+
+    parent->signal_children_list_changed.connect(children_list_change_notifier);
+    child->signal_parent_changed.connect(parent_change_notifier);
+
+    parent->add_child(child);
+    EXPECT_EQ(children_change_count, 1);
+    EXPECT_EQ(parent_change_count, 1);
+
+
```
Comment (arsdever) [inline test/unit/object.cpp]: Extra new line.

File: test/unit/object.cpp [hunk 3]
```diff
@@ -53,3 +53,30 @@ TEST(object, remove)
...
+    g::project::object_ptr new_parent =
+        g::project::object::create("New Parent", nullptr);
+    new_parent->add_child(child);
+    EXPECT_EQ(children_change_count, 2); // Removed 'child' from children. 
+    EXPECT_EQ(parent_change_count, 3);  //Removed old parent. Added new parent.
+
```
Comment (arsdever) [inline test/unit/object.cpp:80]: Please also add the case when the `object` is created with the parent already in the argument list like `create("...", parent);`

**Review by arsdever — CHANGES_REQUESTED** [review]

File: test/unit/object.cpp (hunk 3 above)
Comment (lia-m99) [reply test/unit/object.cpp:80]: Done

File: test/unit/object.cpp (hunk 2 above)
Comment (lia-m99) [reply test/unit/object.cpp]: Done

Comment (arsdever) [reply test/unit/object.cpp]: I still see the new line 🙂

**Comment by arsdever** [comment]
Feels like some commits are missing. Please ensure that everything is pushed.

**Review by arsdever — APPROVED** [review]

**Pull request merged**